### PR TITLE
Implement `valueMap` with `ignoreCase`

### PR DESCRIPTION
**Pull request by alexander-myltsev**
fixes https://github.com/sirthias/parboiled2/issues/184

### CODE REVIEW

**Comment by alexander-myltsev** [comment]
@sirthias , 

do you think `valueMap` with the ignored case should return some particular message in error case?

What was the reason to implement two versions of `__matchString` in sense of case instead of a single function with a boolean flag? If it is about speed, is that really critical to perform `if` branch against `ignoreCase` boolean flag?

File: parboiled-core/src/main/scala/org/parboiled2/Parser.scala [hunk 1]
```diff
@@ -466,15 +467,16 @@ abstract class Parser(initialValueStackSize: Int = 16,
   /**
    * THIS IS NOT PUBLIC API and might become hidden in future. Use only if you know what you are doing!
    */
-  def __matchMapWrapped(m: Map[String, Any]): Boolean = {
+  def __matchMapWrapped(m: Map[String, Any], ignoreCase: Boolean): Boolean = {
+    val matchString = if (ignoreCase) __matchIgnoreCaseStringWrapped _ else __matchStringWrapped _
```
Comment (sirthias) [inline parboiled-core/src/main/scala/org/parboiled2/Parser.scala]: On Java7 (Scala 2.11) this allocates a new function instance _every_ time this method is executed, which can be quite costly.
Pulling the `if` into the `while` should be the better solution. If will be nicely predictable and thus only add negligible overhead.

Comment (alexander-myltsev) [reply parboiled-core/src/main/scala/org/parboiled2/Parser.scala]: I fixed this one. Let me know when you're ready to merge. I'll squash commits beforehand.

**Comment by sirthias** [comment]
Ok, thank you, this looks good now.
The only thing missing is a new test for the added feature...

**Comment by SethTisue** [comment]
seen in Scala 2.12 community build:

```
[parboiled2] [error] /Users/tisue/community.212/target-0.9.10/project-builds/parboiled2-07ec03ac9b0422b56798048546bdf42d3057f133/parboiled-core/src/test/scala/org/parboiled2/BasicSpec.scala:165: Calls to `valueMap` must be inside `rule` macro
[parboiled2] [error]       def targetRule = rule { colors ~ EOI }
[parboiled2] [error]                               ^
[parboiled2] [error] /Users/tisue/community.212/target-0.9.10/project-builds/parboiled2-07ec03ac9b0422b56798048546bdf42d3057f133/parboiled-core/src/test/scala/org/parboiled2/BasicSpec.scala:183: Calls to `valueMap` must be inside `rule` macro
[parboiled2] [error]       def targetRule = rule { map ~ EOI }
[parboiled2] [error]                               ^
[parboiled2] [warn] 8 warnings found
```

related to this PR...?

**Comment by sirthias** [comment]
@SethTisue Yes, thank you, Seth, for the heads-up!
This is indeed a regression that was introduced with this PR.
I just pushed [a commit](https://github.com/sirthias/parboiled2/commit/2923ca4b09f49ad2886a53f441d883449ba50498) with the fix.

**Pull request merged**